tests/test_redis_lock.py

```diff
@@ -499,3 +500,15 @@ def workerfn(unblocked):
     worker.join(0.5)
 
     assert unblocked.value == 1
+
+
+def test_auto_renewal_stops_after_gc(conn):
+    """Auto renewal stops after lock is garbage collected."""
+    lock = Lock(conn, 'spam', auto_renewal=True, expire=1)
+    name = lock._name
+    lock.acquire()
+    del lock
+    gc.collect()
+
+    time.sleep(2)
+    assert conn.get(name) is None
```

Comment:
PyPy docs:
> Last note: CPython tries to do a gc.collect() automatically when the program finishes; not PyPy. (It is possible in both CPython and PyPy to design a case where several gc.collect() are needed before all objects die. This makes CPython’s approach only work “most of the time” anyway.)

Maybe run `gc.collect()` for one second or something? Or configure GC to run more often: http://doc.pypy.org/en/latest/gc_info.html

Not sure what to do, I think some "trial and error" (trying various things) is necessary here.